theano/sandbox/gpuarray/opt.py

```diff
@@ -526,8 +526,9 @@ def local_gpua_softmaxwithbias(node):
 @register_opt('fast_compile')
 @op_lifter([theano.tensor.opt.Assert])
 def local_assert(node):
-    return [host_from_gpu(node.op(node.inputs[0].owner.inputs[0],
-                                  *node.inputs[1:]))]
+    if isinstance(node.inputs[0].owner, HostFromGpu):
```

Comment:
Yeah, plus I did my check wrong, looking at the apply node instead of the op. Don't know what I was thinking. I implemented the right check and did a rebase to keep the history clean.